### PR TITLE
Ch4168/studio monitor replicated yml for changes

### PR DESCRIPTION
**Pull request by laverya**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -13,23 +13,48 @@ import * as process from "process";
...
-  console.log(chalk.green(`Watching "${consts.localPath}" for changes. Any updates to this file will be sent as an application update.`));
+  console.log(chalk.green(`Watching "${path.join(consts.localPath, "current.yaml")}" for changes. Any updates to this file will be sent as an application update.`));
+  fs.watchFile(path.join(consts.localPath, "current.yaml"), (curr, prev) => {
+    const releases = listAvailableReleases();
+    if (releases.length === 0) {
+      fs.copyFileSync(path.join(consts.localPath, "current.yaml"), path.join(consts.localPath, "releases", "1.yaml"));
+      console.log("Created first release at releases/1.yaml");
+    } else {
+      // next release is at the number of last, plus 1
+      const last = releases[releases.length - 1];
```
Comment (divolgin) [inline src/index.ts]: So if there is 1 release, next one becomes 0, and if there are 2, we are back at 1?

File: src/index.ts [hunk 2]
```diff
@@ -13,23 +13,48 @@ import * as process from "process";
...
+      let name = "";
+      if (last.indexOf(".yaml") !== -1) {
```
Comment (divolgin) [inline src/index.ts]: I'd split this on `.` and and take first part as `name`

File: src/replicated/helpers.ts [hunk 3]
```diff
@@ -1,4 +1,46 @@
...
+        console.log(chunk);
+        yamldata = yamldata + chunk.toString();
+      });
+
+      newRes.on("end", () => {
+        console.log("end");
+        fs.writeFileSync(path.join(consts.localPath, "releases", sequence + ".yaml"), yamldata);
+        if (!fs.existsSync(path.join(consts.localPath, "current.yaml"))) {
+          fs.copyFileSync(path.join(consts.localPath, "releases", sequence + ".yaml"), path.join(consts.localPath, "current.yaml"));
+          watch();
```
Comment (divolgin) [inline src/replicated/helpers.ts]: this looks like it creates multiple watches for the same file

File: src/index.ts (hunk 1 above)
Comment (laverya) [reply src/index.ts]: If there is 1 release, pick the one at index 0 - the releases are sorted, and we pick the last one

File: src/replicated/helpers.ts (hunk 3 above)
Comment (laverya) [reply src/replicated/helpers.ts]: It only creates a watch if current.yaml had not existed beforehand, and the watch code exits if current.yaml doesn't exist
So only one watch will be running at a time

File: src/index.ts (hunk 2 above)
Comment (laverya) [reply src/index.ts]: Yeah, that sounds like a better idea

**Pull request merged**